src/common/runtime/src/repeated_task.rs

```diff
@@ -99,15 +107,17 @@ impl<E: ErrorExt + 'static> RepeatedTask<E> {
             IllegalStateSnafu { name: &self.name }
         );
 
-        let interval = self.interval;
         let child = self.cancel_token.child_token();
         // Safety: The task is not started.
         let mut task_fn = inner.task_fn.take().unwrap();
+        let interval = self.interval;
+        let mut initial_delay = self.initial_delay;
         // TODO(hl): Maybe spawn to a blocking runtime.
         let handle = runtime.spawn(async move {
             loop {
+                let sleep_time = initial_delay.take().unwrap_or(interval);
```

Comment:
What about checking if the `sleep_time` is greater than zero? If not, skip the `tokio::select!`? Avoid the invocation.

Reply:
good idea